backend/Backend.mo

```diff
@@ -193,14 +193,29 @@ actor class Backend(_owner : Principal) {
     return true;
   };
 
-    public shared ({ caller }) func requestCreatorAccess(details : Types.CreatorApproval) : async Bool {
+  public shared ({ caller }) func requestCreatorAccess(details : Types.CreatorApproval) : async Bool {
     if (details.principal != caller) {
-      throw Error.reject("Principal does not match");
+        throw Error.reject("Principal does not match");
     };
 
-    creatorPendingApproval.add(details);
-    return true;
-  };
+    let existingApproval = Array.find(
+        Buffer.toArray(creatorPendingApproval),
+        func(approval : Types.CreatorApproval) : Bool {
+            approval.principal == details.principal;
+        },
+    );
+
+    switch (existingApproval) {
+        case (?approval) {
+            throw Error.reject("Approval request already exists for this user");
+        };
+        case null {
+            creatorPendingApproval.add(details);
```

Comment:
CreatedAt and UpdateAt should be added here not from the frontend

Reply:
actually I have updated the data structure of createrApproval. Wont that suffice to the requirements?

<img width="533" alt="Screenshot 2024-07-04 at 6 17 53 PM" src="https://github.com/elna-ai/ELNA-DApp/assets/30034816/93642d3b-c717-4c87-82f1-7979bb082b71">